pkg/api/types.go

```diff
@@ -2070,10 +2070,41 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// These are the valid conditions for the component type.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
+	ComponentScheduler         ComponentType = "scheduler" // TODO: move mesos-specific component type & validation to contrib
+)
+
+// Component describes metadata about a core component.
+type Component struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Type of the component
+	Type ComponentType `json:"type"`
+	// URL to reach the component (scheme, host, port, path)
```

Comment:
For now all components are expected to have a `/healthz` endpoint, regardless of type. But this may change if we register etcd as a component. 

Actually, I added this location url before switching to heartbeats. It's not currently being used except to display to the user.